src/LayoutManager/LayoutStrategies/FitContentLayout.ts

```diff
@@ -0,0 +1,50 @@
+import type { Point } from '../../Point';
+import type { Group } from '../../shapes/Group';
+import type { FabricObject } from '../../shapes/Object/FabricObject';
+import { makeBoundingBoxFromPoints } from '../../util/misc/boundingBoxFromPoints';
+import type { LayoutStrategyResult, StrictLayoutContext } from '../types';
+import { LayoutStrategy } from './LayoutStrategy';
+import { getObjectBounds, getObjectSizeVector } from './utils';
+
+export class FitContentLayout extends LayoutStrategy {
+  /**
+   * optimize layout when possible
```

Comment:
Should explain what's exactly is delayed. Code should always make clear "what" and "why".

Reply:
I will add comments
I was thinking of this use case after our chat today...
Adding scaling, rotating, resizing, moving triggers makes sense to me.
In fact I want to move the trigger logic to the manager, to an intialize/attach method so you can also fine tune them.
Back to topic - if we add these triggers we can use the lazy concept to calc layout of the transforming object and cache the rest of the objects' bbox for quick computation.

Reply:
you think `optimize`` is a better name??